### PR TITLE
fix: internal schema apprears as object on logout

### PR DESCRIPTION
**Pull request by SB-harshitajadhav**
[STOP-1027](https://smartbear.atlassian.net/browse/STOP-1027)

In general, make sure you have: (check the boxes to acknowledge you've followed this template)

Previously, if not logged in, the schema which is marked internal showing up in dropdown as an empty 'object' for oneOf/anyOf in public docs.

### Before 

image (13)
image (14)

### After 

image (15)
image (16)

In case of every schema is internal in the list it will display the error message.

image (18)

- [ ] Read [`CONTRIBUTING.md`](../CONTRIBUTING.md)

#### Other Available PR Templates:

- Release: https://github.com/stoplightio/elements/compare?template=release.md
  - [ ] [Read the release section of `CONTRIBUTING.md`](../CONTRIBUTING.md#releasing-elements)

[STOP-1027]: https://smartbear.atlassian.net/browse/STOP-1027?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/elements-core/src/utils/ref-resolving/resolvedObject.ts [hunk 1]
```diff
@@ -78,7 +78,30 @@ export const isResolvedObjectProxy = (someObject: object): boolean => {
 };
 
 export const getOriginalObject = (resolvedObject: object): object => {
-  return (resolvedObject as Record<symbol, object>)[originalObjectSymbol] || resolvedObject;
+  const originalObject: any = (resolvedObject as Record<symbol, object>)[originalObjectSymbol] || resolvedObject;
```
Comment (bhaskarsontakke) [inline packages/elements-core/src/utils/ref-resolving/resolvedObject.ts:81]: Is the function also handle the allOf usecase ?

Comment (SB-harshitajadhav) [reply packages/elements-core/src/utils/ref-resolving/resolvedObject.ts:81]: In allOf it is not showing empty object it is working as expected so just handled anyOf and oneOf.

**Review by bhaskarsontakke — APPROVED** [review]
looks good to me., though I have added on comment. Please address it.

**Review by ben-smartbear — CHANGES_REQUESTED** [review]
going to need some test coverage for this

**Review by SB-akshaythakar — APPROVED** [review]

File: packages/elements-core/src/utils/ref-resolving/resolvedObject.test.ts [hunk 2]
```diff
@@ -236,4 +236,53 @@ describe('createResolvedObject', () => {
 
     expect(resolvedObject).toEqual(originalObject);
   });
+
+  it('removes object if contains an error for oneOf', () => {
+    const originalObject = {
```
Comment (bhaskarsontakke) [inline packages/elements-core/src/utils/ref-resolving/resolvedObject.test.ts:241]: you can define originalObject and filteredObject at top. So, it won't be repeating in the test suit

File: packages/elements-core/src/utils/ref-resolving/resolvedObject.test.ts [hunk 3]
```diff
@@ -236,4 +236,176 @@ describe('createResolvedObject', () => {
...
+              id: 'nezai0hyj4yak',
+            },
+          },
+        ],
+      ],
+      'x-stoplight': { id: 'b73ff5df9864f' },
+    };
+    expect(resolvedObject).toEqual(filteredObject);
+  });
+  it('show the error if all object contains an error for oneOf', () => {
```
Comment (bhaskarsontakke) [inline packages/elements-core/src/utils/ref-resolving/resolvedObject.test.ts:350]: may be you can put a comment saying "The schema marked as internal so it's populate the error object.

**Review by bhaskarsontakke — APPROVED** [review]
Changes looks to good to me

**Review by SB-akshaythakar — APPROVED** [review]

**Review by ben-smartbear — APPROVED** [review]

**Pull request merged**